core/src/main/scala/org/apache/spark/rdd/PipedRDD.scala

```diff
@@ -157,8 +167,16 @@ private[spark] class PipedRDD[T: ClassTag](
     val lines = Source.fromInputStream(proc.getInputStream).getLines()
     new Iterator[String] {
       def next(): String = lines.next()
+
+      private def propagateChildThreadException(): Unit = {
```

Comment:
Why a method here? actually this whole iterator implementation looks like it needs some work. `hasNext` is supposed to be idempotent, but it's doing the work of checking the status. You'd never see an exception if you never called it.

Why not check the result of the process exit before returning the iterator? then you just return an iterator from `lines`. This is already effectively blocking on the return from the process anyway.